Sources/Fluent/Database/ThreadConnectionPool.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/**
+    Responsible for maintaing of pool
+    of connections, one for each thread.
+ 
+    Uses a Driver to make new connections.
+*/
+public final class ThreadConnectionPool {
+    var makeConnection: ConnectionFactory
+    var storage: [pthread_t: (conn: Connection, time: Int)]
```

Comment:
> Return a connection to the pool when the request is complete. 

That's exactly what we want to happen but
- we'd have to couple fluent more closely with vapor to know when the request is over or not
- that logic wouldn't really apply to projects that might use fluent in a different context

A "one connection per thread" policy with a rule to start re-using old connections once a max is reached seems like it could be general enough to work for all use cases.